libraries/AC_AutoTune/AC_AutoTune_Multi.h

```diff
@@ -28,172 +28,194 @@
-    }
-
-    // update gains for the angle P up tune type
     void updating_angle_p_up_all(AxisType test_axis) override;
-
-    // update gains for the angle P down tune type
     void updating_angle_p_down_all(AxisType test_axis) override;
 
-    // update gains for the max gain tune type
-    void updating_max_gains_all(AxisType test_axis) override {
-        // this should never happen
+    // These tune types are not used in multirotors
+    void updating_rate_ff_up_all(AxisType) override {
+        INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
+    }
+    void updating_max_gains_all(AxisType) override {
         INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
     }
 
-    // set gains post tune for the tune type
-    void set_gains_post_tune(AxisType test_axis) override;
+    // Applies gain margin (backoff) at the end of a TuneType
+    void set_tuning_gains_with_backoff(AxisType test_axis) override;
 
-    // reverse direction for twitch test
+    // Reverses twitch direction for next test
     bool twitch_reverse_direction() override { return !positive_direction; }
 
 #if HAL_LOGGING_ENABLED
     void Log_AutoTune() override;
     void Log_AutoTuneDetails() override;
     void Log_AutoTuneSweep() override {
-        // this should never happen
         INTERNAL_ERROR(AP_InternalError::error_t::flow_of_control);
     }
-    void Log_Write_AutoTune(AxisType axis, uint8_t tune_step, float meas_target, float meas_min, float meas_max, float new_gain_rp, float new_gain_rd, float new_gain_sp, float new_ddt);
+
+    void Log_Write_AutoTune(AxisType axis, TuneType tune_step,
+                            float meas_target, float meas_min, float meas_max,
+                            float new_gain_rp, float new_gain_rd, float new_gain_sp, float new_ddt);
     void Log_Write_AutoTuneDetails(float angle_cd, float rate_cds);
 #endif
 
+    // Defines the sequence of tuning steps (P, D, etc.)
     void set_tune_sequence() override {
-        tune_seq[0] = RD_UP;
-        tune_seq[1] = RD_DOWN;
-        tune_seq[2] = RP_UP;
-        tune_seq[3] = SP_DOWN;
-        tune_seq[4] = SP_UP;
-        tune_seq[5] = TUNE_COMPLETE;
+        tune_seq[0] = TuneType::RATE_D_UP;
+        tune_seq[1] = TuneType::RATE_D_DOWN;
+        tune_seq[2] = TuneType::RATE_P_UP;
+        tune_seq[3] = TuneType::ANGLE_P_DOWN;
+        tune_seq[4] = TuneType::ANGLE_P_UP;
+        tune_seq[5] = TuneType::TUNE_COMPLETE;
     }
 
-    // get_axis_bitmask accessor
+    // Axis tuning mask accessor
     uint8_t get_axis_bitmask() const override { return axis_bitmask; }
 
-    // get_testing_step_timeout_ms accessor
+    // Timeout for a single twitch test
     uint32_t get_testing_step_timeout_ms() const override;
 
 private:
-    // twitch test functions for multicopter
-    void twitch_test_init();
-    void twitch_test_run(AxisType test_axis, const float dir_sign);
+    // Helpers for twitch-based test monitoring
+
```

Comment:
white space is a bit odd from here on down.  a number of empty lines here and then lower down all the methods are all left justified